client/search/search-input.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import SearchIcon from '../icons/material/baseline-search-24px.svg'
+import { useKeyListener } from '../keyboard/key-listener'
+import { fastOutSlowInShort } from '../material/curves'
+import TextField from '../material/text-field'
+import { useValueAsRef } from '../state-hooks'
+
+const ESCAPE = 'Escape'
+const F = 'KeyF'
+
+const TextFieldContainer = styled(TextField)`
+  width: ${props => (props.isFocused ? '250px' : '200px')};
+  ${fastOutSlowInShort};
+`
+
+interface SearchInputProps {
+  searchQuery: string
+  onSearchChange: (value: string) => void
+  className?: string
+}
+
+export function SearchInput({ searchQuery, onSearchChange, className }: SearchInputProps) {
+  const [inputValue, setInputValue] = useState('')
+  const [inputFocused, setInputFocused] = useState(false)
+  const inputFocusedRef = useValueAsRef(inputFocused)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    // In case the `searchQuery` (the value that was actually searched for) gets out of sync with
+    // the input value (which can happen if network requests return out of order for example), we
+    // update the input value to what was actually searched for so it reflects the search results
+    // accurately.
+    setInputValue(searchQuery)
```

Comment:
If you initialize it properly as I said above, this shouldn't ever be necessary. What was happening in your testing wasn't "network requests returned out of order", it's that you weren't initialized the state to the actual value, so until a change event was delivered by the input it never got updated

Reply:
This absolutely should be handled, but I'm not sure why you'd be handling it here or in this way. That particular ordering should already be handled in the action creator, which only dispatches actions for the most recent request's response